theano/tensor/signal/downsample.py

```diff
@@ -845,8 +979,6 @@ def make_node(self, x, maxout, gz):
         return Apply(self, [x, maxout, gz], [x.type()])
 
     def perform(self, node, inp, out):
-        if self.mode != 'max':
-            raise theano.gof.utils.MethodNotDefined()
```

Comment:
As the perform only support max, for safety, add an assert in the \__init__ as you removed it here.